### PR TITLE
refine sentiment en_doc

### PR DESCRIPTION
**Pull request by gangliao**

### CODE REVIEW

File: understand_sentiment/README.en.md [hunk 1]
```diff
@@ -108,377 +114,248 @@ As shown in Figure 4 (3-layer RNN), odd/even layers are forward/reverse LSTM. Hi
...
 
-```
-Loading parameters from model_output/pass-00002/
-predicting label is pos
-```
+## Conclusion
 
-`10007_10.txt` in folder`./data/aclImdb/test/pos`, the predicted label is also pos，so the prediction is correct.
-## Summary
 In this chapter, we use sentiment analysis as an example to introduce applying deep learning models on end-to-end short text classification, as well as how to use PaddlePaddle to implement the model. Meanwhile, we briefly introduce two models for text processing: CNN and RNN. In following chapters we will see how these models can be applied in other tasks.
```
Comment (helinwang) [inline understand_sentiment/README.en.md:355]: `In following chapters` -> `In following chapters, `

File: understand_sentiment/index.en.html [hunk 2]
```diff
@@ -150,377 +156,248 @@
...
-```bash
-./data/get_imdb.sh
-```
-If successful, you should see the directory ```data``` with following files:
+`paddle.datasets` package encapsulates multiple public datasets, including `cifar`, `imdb`, `mnist`, `moivelens` and `wmt14`, etc. There's no need for us to manually download and preprocess IMDB.
 
-```
-aclImdb  get_imdb.sh  imdb  mosesdecoder-master
-```
+After issuing a command `python train.py`, training will starting immediately. The details will be unpacked by the following sessions to see how it works.
```
Comment (helinwang) [inline understand_sentiment/index.en.html:165]: `training will starting` -> `training will start`

File: understand_sentiment/index.en.html [hunk 3]
```diff
@@ -150,377 +156,248 @@
...
-We taks the [IMDB sentiment analysis dataset](http://ai.stanford.edu/%7Eamaas/data/sentiment/) as an example. IMDB dataset contains training and testing set, with 25000 movie reviews. With a 1-10 score, negative reviews are those with score<=4, while positives are those with score>=7. You may use following scripts to download the IMDB dataset and [Moses](http://www.statmt.org/moses/) toolbox:
+## Dataset
 
+We use [IMDB](http://ai.stanford.edu/%7Eamaas/data/sentiment/) dataset for sentiment analysis in this tutorial, which consists of 50,000 movie reviews split evenly into 25k train and 25k test sets. In the labeled train/test sets, a negative review has a score <= 4 out of 10, and a positive review has a score >= 7 out of 10.
 
-```bash
-./data/get_imdb.sh
-```
-If successful, you should see the directory ```data``` with following files:
+`paddle.datasets` package encapsulates multiple public datasets, including `cifar`, `imdb`, `mnist`, `moivelens` and `wmt14`, etc. There's no need for us to manually download and preprocess IMDB.
```
Comment (helinwang) [inline understand_sentiment/index.en.html:163]:
```
`moivelens` and
```
->
```
`moivelens`, and
```

File: understand_sentiment/README.en.md [hunk 4]
```diff
@@ -108,377 +114,248 @@ As shown in Figure 4 (3-layer RNN), odd/even layers are forward/reverse LSTM. Hi
...
+
+    Parameter `input_dim` denotes the dictionary size, and `class_dim` is the number of categories. In `stacked_lstm_net`, the input to the network is defined in `paddle.layer.data`.
+
+1. Define Classifier
+
+    The above stacked bidirectional LSTM network extracts high-level features and maps them to a vector of the same size as the categories. `paddle.activation.Softmax` function or classifier is then used for calculating the probability of the sentence belonging to each category.
+
+1. Define Loss Function
+
+    In the context of supervised learning, labels of training set are defined in `paddle.layer.data`, too. During training, cross-entropy is used as loss function in `paddle.layer.classification_cost` and as the output of the network; During testing, the outputs are the probabilities calculated in the classifier.
```
Comment (helinwang) [inline understand_sentiment/README.en.md:260]: training set -> the training set

File: understand_sentiment/README.en.md [hunk 5]
```diff
@@ -108,377 +114,248 @@ As shown in Figure 4 (3-layer RNN), odd/even layers are forward/reverse LSTM. Hi
...
-else:
-outputs(output)
-```
+    The above Text CNN network extracts high-level features and maps them to a vector of the same size as the categories. `paddle.activation.Softmax` function or classifier is then used for calculating the probability of the sentence belonging to each category.
 
-In our implementation, we can use just a single layer [`sequence_conv_pool`](https://github.com/PaddlePaddle/Paddle/blob/develop/python/paddle/trainer_config_helpers/networks.py) to do convolution and pooling operation, convolution kernel size set as hidden_size parameters.
+1. Define Loss Function
 
-#### Implementation of Stacked bidirectional LSTM
+    In the context of supervised learning, labels of training set are defined in `paddle.layer.data`, too. During training, cross-entropy is used as loss function in `paddle.layer.classification_cost` and as the output of the network; During testing, the outputs are the probabilities calculated in the classifier.
```
Comment (helinwang) [inline understand_sentiment/README.en.md:175]: training set -> the training set

File: understand_sentiment/index.en.html [hunk 6]
```diff
@@ -150,377 +156,248 @@
...
 
-```python
-settings(
-batch_size=128,
-learning_rate=2e-3,
-learning_method=AdamOptimizer(),
-regularization=L2Regularization(8e-4),
-gradient_clipping_threshold=25)
-```
+    Parameter `input_dim` denotes the dictionary size, and `class_dim` is the number of categories. In `convolution_net`, the input to the network is defined in `paddle.layer.data`.
```
Comment (helinwang) [inline understand_sentiment/index.en.html]: is defined in -> is defined using

File: understand_sentiment/README.en.md (hunk 5 above)
Comment (helinwang) [inline understand_sentiment/README.en.md]: are defined by

File: understand_sentiment/README.en.md (hunk 4 above)
Comment (helinwang) [inline understand_sentiment/README.en.md]: are defined in -> are defined by

File: understand_sentiment/README.en.md [hunk 7]
```diff
@@ -108,377 +114,248 @@ As shown in Figure 4 (3-layer RNN), odd/even layers are forward/reverse LSTM. Hi
...
 
-```
-Pass=0 samples=24999 AvgCost=0.280471 Eval: classification_error_evaluator=0.115645
+```python
+feeding = {'word': 0, 'label': 1}
 ```
 
-### Prediction
-`predict.py` script provides an API. Predicting IMDB data without labels as following:
+Callback function `event_handler` will be invoked to track training and testing process when a pre-defined event happens.
```
Comment (helinwang) [inline understand_sentiment/README.en.md:325]: testing不会调用callback吧？
track training and testing process -> track training progress (progress不是process哈～)

File: understand_sentiment/README.en.md [hunk 8]
```diff
@@ -108,377 +114,248 @@ As shown in Figure 4 (3-layer RNN), odd/even layers are forward/reverse LSTM. Hi
...
+```python
+trainer.train(
+    reader=train_reader,
+    event_handler=event_handler,
+    feeding=feedig,
+    num_passes=10)
+```
 
-Prediction result of our example:
+After training is done, the model from each pass is saved in `output/pass-%05d`. For example, the model of Pass 300 is saved in `output/pass-00299`.
```
Comment (helinwang) [inline understand_sentiment/README.en.md:351]: 这个`event_handler`里并没有存模型的代码，所以应该不会有`output/pass-%05d`。这一行还是删掉吧。

**Review by helinwang — APPROVED** [review]
LGTM. I will send a follow up PR to fix these issues.

File: understand_sentiment/README.en.md (hunk 5 above)
Comment (helinwang) [reply understand_sentiment/README.en.md:175]: Done.

File: understand_sentiment/README.en.md (hunk 4 above)
Comment (helinwang) [reply understand_sentiment/README.en.md:260]: Done.

File: understand_sentiment/README.en.md (hunk 7 above)
Comment (helinwang) [reply understand_sentiment/README.en.md:325]: Done.

File: understand_sentiment/README.en.md (hunk 8 above)
Comment (helinwang) [reply understand_sentiment/README.en.md:351]: Done.

File: understand_sentiment/README.en.md (hunk 1 above)
Comment (helinwang) [reply understand_sentiment/README.en.md:355]: Done.

File: understand_sentiment/index.en.html (hunk 3 above)
Comment (helinwang) [reply understand_sentiment/index.en.html:163]: Done.

File: understand_sentiment/index.en.html (hunk 2 above)
Comment (helinwang) [reply understand_sentiment/index.en.html:165]: Done.

**Comment by helinwang** [comment]
Fixes are in https://github.com/PaddlePaddle/book/pull/193
